### PR TITLE
[docs] Add a recipe describing integration with Azure DevOps

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**
\cc @DevExpress/testcafe-docs

### CODE REVIEW

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+title: Integrating TestCafe with Azure DevOps
+permalink: /documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.html
+---
+# Integrating TestCafe with Azure DevOps
+
+This topic describes how to integrate TestCafe tests into an [Azure DevOps Server](https://azure.microsoft.com/en-us/services/devops/server/) and [Azure Pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/) project's build process.
+
+## Step 1 - Add TestCafe Dependency to the Project
+
+Open `package.json` or create it in the repository root. Add `testcafe` to the `devDependencies` section.
```
Comment (AndreyBelym) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: IMHO it's better to use 
```
npm --save-dev testcafe
```
to add TestCafe as a dev dependency.

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+```json
+{
+    "devDependencies": {
+        "testcafe": "^0.23.2"
```
Comment (AndreyBelym) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: I don't feel that hardcoding the TestCafe version in the recipe is a good idea.

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+![Create New Task](../../../images/azure/add-new-task.png)
+
+Select **Node Tool Installer** from the task list. This task installs Node.js and adds it to the `PATH` variable.
+
+![Add Node.js Installer](../../../images/azure/add-node-installer.png)
+
+The task is now added to the pipeline. Click its name in the pipeline to enter the task settings.
+
+On the settings screen, specify the Node.js version to install - `8.x`.
```
Comment (AndreyBelym) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: A specific Node.js version, can we avoid using it?

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+## Step 2 - Provide a Command to Run Tests
+
+Since TestCafe is installed locally, a test run command that launches TestCafe should be also added to `package.json` to the `scripts` section.
+
+```json
+{
+    "scripts": {
+        "test": "testcafe chrome:headless tests/**/*"
+    },
+    "devDependencies": {
```
[the remark this answers is not in the document]
Comment (helen-dikareva) [reply docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: ` some screenshots the real data ` without any further explanation could make you think that you need to use a particular version

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
+---
+layout: docs
+title: Integrating TestCafe with Azure DevOps
+permalink: /documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.html
+---
+# Integrating TestCafe with Azure DevOps
+
+This topic describes how to integrate TestCafe tests into an [Azure DevOps Server](https://azure.microsoft.com/en-us/services/devops/server/) and [Azure Pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/) project's build process.
```
Comment (helen-dikareva) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: Should we add some note for those guys who asked us about TFS integration (why we provide this recipe instead)?

Comment (VasilyStrelyaev) [reply docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: Well, I begin thinking that we need a recipe for TFS as well.

Comment (AlexSkorkin) [reply docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: :-)

Probably, we should describe why we don't have a recipe for TFS (just to add search keywords for those who look for such a recipe).

Comment (VasilyStrelyaev) [reply docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: Even if we never write about TFS, how is this explanation going to help people who use TFS?

**Comment by VasilyStrelyaev** [comment]
@testcafe-build-bot \retest

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+{
+    "devDependencies": {
+        "testcafe": "*"
+    }
+}
+```
+
+## Step 2 - Provide a Command to Run Tests
+
+Since TestCafe is installed locally, a test run command that launches TestCafe should be also added to `package.json` to the `scripts` section.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: should be also added? Do you mean 'should also be added'?

Comment (dirk-pieterse) [reply docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: Same change to this sentence as in your other doc.

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+        "test": "testcafe chrome:headless tests/**/*"
+    },
+
+    "devDependencies": {
+        "testcafe": "*"
+    }
+}
+```
+
+This allows you to run tests using `npm test`.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: Same here.

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+      versionSpec: '10.14'
+    displayName: 'Install Node.js'
+  - script: npm install
+    displayName: 'Install TestCafe'
+  - script: npm test
+    displayName: 'Run TestCafe Tests'
+```
+
+* `jobs` - the list of [jobs](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=vsts&tabs=yaml).
+* `job` - the job name. You can choose any name you wish.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: Why do you add 'You can choose any name you wish'?

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+* `pool` - the [agent pool](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/pools-queues?view=vsts) that executes this build job.
+* `vmImage` - the [agent's](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/agents?view=vsts) virtual machine image name. This tutorial uses a [Microsoft-hosted agent](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=vsts&tabs=yaml) that runs on an Ubuntu 16.04 machine. You can find the list of software installed on this machine in the [Azure Pipelines GitHub repository](https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/linux/Ubuntu1604-README.md).
+* `steps` - the list of steps performed when executing a job.
+* `task` - adds a [Node.js installer task](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/node-js?view=vsts)  to the job. This task installs Node.js and adds it to the `PATH` variable.
+* `inputs` - specifies input variables for the task.
+* `versionSpec` - the Node.js version to install. You can find the latest LTS version number on the [Node.js website](https://nodejs.org/en/).
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: You can find the latest LTS version number on -> The latest LTS version number is available on

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+
+### Option 2 - Run the Build Manually
+
+To run the build manually, select the build and click **Queue**.
+
+On Azure DevOps Server, you can find this button under the build header.
+
+![Queue Task - Server](../../../images/azure/queue-task-server.png)
+
+On Azure Pipelines, this button is located in the top-right corner.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: without 'located'

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+
+![Triggers Tab](../../../images/azure/enable-ci.png)
+
+If you need to build a specific branch or path, you can add **Branch filters** and/or **Path filters**.
+
+Now the build is triggered by code changes in the repository.
+
+### Option 2 - Run the Build Manually
+
+To run the build manually, select the build and click **Queue**.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: Select the build and click **Queue** to run the build.

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+
+![Triggers Tab](../../../images/azure/set-ci.png)
+
+Click the repository name under the **Continuous integration** header and check **Enable continuous integration**.
+
+![Triggers Tab](../../../images/azure/enable-ci.png)
+
+If you need to build a specific branch or path, you can add **Branch filters** and/or **Path filters**.
+
+Now the build is triggered by code changes in the repository.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: Now, changes to code in the repository trigger the build.

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+
+Then open the **Triggers** tab.
+
+![Triggers Tab](../../../images/azure/set-ci.png)
+
+Click the repository name under the **Continuous integration** header and check **Enable continuous integration**.
+
+![Triggers Tab](../../../images/azure/enable-ci.png)
+
+If you need to build a specific branch or path, you can add **Branch filters** and/or **Path filters**.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: If you need to build a specific branch or path, you can add **Branch filters** and/or **Path filters** -> To build a specific branch or path, add **Branch filters** and/or **Path filters**.

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+
+This opens the **Save build pipeline and queue** window. Specify an optional comment that describes the changes you have made. Select an agent pool that runs this task, the repository branch and the commit that contains code to build. You can leave the **Commit** field blank to use the latest commit. Click **Save & queue**.
+
+![Save Pipeline](../../../images/azure/save-options.png)
+
+## Step 4 - Run Test Task and View Results
+
+### Option 1 - Enable Continuous Integration
+
+You can set Azure DevOps to run the pipeline every time your code changes. To do this, you need to enable continuous integration.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: You can enable continuous integration in Azure DevOpst to specify that the pipeline is run every time your code changes?

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+
+On the settings screen, specify the Node.js version to install. Use the latest LTS version - you can find its number on the [Node.js website](https://nodejs.org/en/).
+
+![Specify Node.js Version](../../../images/azure/set-node-version.png)
+
+Now add another task that runs a shell command. Click the **+** button again and select **Command Line** from the task list.
+
+![Add a Command Line Task](../../../images/azure/add-command-line.png)
+
+Configure this command to install TestCafe. Open its settings, specify a meaningful name and enter the following command:
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: a meaningful name?

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 16]
```diff
@@ -0,0 +1,204 @@
...
+
+![Create New Task](../../../images/azure/add-new-task.png)
+
+Select **Node Tool Installer** from the task list. This task installs Node.js and adds it to the `PATH` variable.
+
+![Add Node.js Installer](../../../images/azure/add-node-installer.png)
+
+The task is now added to the pipeline. Click its name in the pipeline to enter the task settings.
+
+On the settings screen, specify the Node.js version to install. Use the latest LTS version - you can find its number on the [Node.js website](https://nodejs.org/en/).
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: See previous comment

File: docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md [hunk 17]
```diff
@@ -0,0 +1,204 @@
...
+* `displayName` - a step name displayed in build results.
+* `script` - a console command executed at this step.
+
+The first step in this task installs TestCafe with `npm install` and the second step runs TestCafe tests using `npm test`.
+
+Commit your changes and push them to the repository.
+
+### Configure Build Task on Azure DevOps Server
+
+Hover over the **Pipelines** section and click **Builds** in the popup menu.
```
Comment (dirk-pieterse) [inline docs/articles/documentation/recipes/integrating-testcafe-with-ci-systems/azure-devops.md]: pop-up

**Pull request merged**